### ISSUE TITLE
HELP WANTED: MNIST Error

### ISSUE THREAD

**Issue by warf34** [kubeflow/katib#670]
Hello, I would like to test the MNIST code with katib.
I tested it with the code below, but it does not produce any results.
What is the problem? Please tell me why.

p.s. It works very well in the docker.

mnist.yaml
```
apiVersion: "kubeflow.org/v1alpha1"
kind: StudyJob
metadata:
  namespace: kubeflow
  labels:
    controller-tools.k8s.io: "1.0"
  name: mnist
spec:
  studyName: mnist
  owner: ylim
  optimizationtype: maximize
  objectivevaluename: "A"
  optimizationgoal: 1.0
  requestcount: 2
  metricsnames:
  parameterconfigs:
    - name: --t
      parametertype: double
      feasible:
        min: "0.1"
        max: "0.9"
  workerSpec:
    goTemplate:
        rawTemplate: |-
          apiVersion: batch/v1
          kind: Job
          metadata:
            name: {{.WorkerID}}
            namespace: kubeflow
          spec:
            template:
              spec:
                containers:
                - name: {{.WorkerID}}
                  image: radics93/mnist:1.0
                  command:
                  - "python"
                  - "mnist.py"
                  {{- with .HyperParameters}}
                  {{- range .}}
                  - "{{.Name}}={{.Value}}"
                  {{- end}}
                  {{- end}}
                restartPolicy: Never
  suggestionSpec:
    suggestionAlgorithm: "random"
    requestNumber: 2
```

mnist.py
```
import tensorflow as tf
from tensorflow.examples.tutorials.mnist import input_data
mnist = input_data.read_data_sets("MNIST_data/", one_hot=True)

import argparse

parser = argparse.ArgumentParser()
parser.add_argument("--t", type=float, help="???")
args = parser.parse_args()

t = args.t

nb_classes = 10

# MNIST data image of shape 28 * 28 = 784
X = tf.placeholder(tf.float32, [None, 784])
# 0 - 9 digits recognition = 10 classes
Y = tf.placeholder(tf.float32, [None, nb_classes])

W = tf.Variable(tf.random_normal([784, nb_classes]))
b = tf.Variable(tf.random_normal([nb_classes]))

# Hypothesis (using softmax)
hypothesis = tf.nn.softmax(tf.matmul(X, W) + b)

cost = tf.reduce_mean(-tf.reduce_sum(Y * tf.log(hypothesis), axis=1))
train = tf.train.GradientDescentOptimizer(learning_rate=t).minimize(cost)

# Test model
is_correct = tf.equal(tf.argmax(hypothesis, 1), tf.argmax(Y, 1))
# Calculate accuracy
accuracy = tf.reduce_mean(tf.cast(is_correct, tf.float32))

# parameters
num_epochs = 15
batch_size = 100
num_iterations = int(mnist.train.num_examples / batch_size)

sess = tf.Session()
# Initialize TensorFlow variables
sess.run(tf.global_variables_initializer())
# Training cycle
for epoch in range(num_epochs):
    avg_cost = 0

    for i in range(num_iterations):
        batch_xs, batch_ys = mnist.train.next_batch(batch_size)
        _, cost_val = sess.run([train, cost], feed_dict={X: batch_xs, Y: batch_ys})
        avg_cost += cost_val / num_iterations

# Test the model using test sets
A = accuracy.eval(session=sess, feed_dict={X: mnist.test.images, Y: mnist.test.labels})
sess.close()
print("Accuracy: %f" % A)
```

Dockerfile
```
FROM python:3.6
FROM tensorflow/tensorflow

MAINTAINER "AAA"

ENV PYTHONUNBUFFERED=0

ADD . ./
WORKDIR ./
```

**Comment by johnugeorge** [comment]
Can you look at logs of study job controller?

**Comment by warf34** [comment]
How do I look the logs of study job controllers?

**Comment by warf34** [comment]
> Can you look at logs of study job controller?
I found this log.
```
2019/06/25 17:44:15 Adding finalizer clean-studyjob-data
2019/06/25 17:44:15 Start to Validate Suggestion Parameters
2019/06/25 17:44:15 Suggestion Parameters are valid
2019/06/25 17:44:15 Create Study mnist
2019/06/25 17:44:15 Study ID b3e937c9b4e9a281
2019/06/25 17:44:15 Study ID b3e937c9b4e9a281 StudyConf name:"mnist" owner:"ylim" optimization_type:MAXIMIZE optimization_goal:1 parameter_configs:<configs:<name:"--t" parameter_type:DOUBLE feasible:<max:"0.9" min:"0.1" > > > objective_value_name:"A" metrics:"A" jobId:"d9b9d5d3-9770-11e9-80b0-080027bee937" job_type:"HP" 
2019/06/25 17:44:15 Study: b3e937c9b4e9a281 Suggestion Spec &{random [] 2}
2019/06/25 17:44:15 Study: b3e937c9b4e9a281 setSuggesitonParameterReply param_id:"sfb40a0d3fed3d90" 
2019/06/25 17:44:15 Study: b3e937c9b4e9a281 CreatedTrials :
2019/06/25 17:44:15 	trial_id:"d9e01b606dce0b7a" study_id:"b3e937c9b4e9a281" parameter_set:<name:"--t" parameter_type:DOUBLE value:"0.4861" > 
2019/06/25 17:44:15 	trial_id:"q5c9d35a8f988475" study_id:"b3e937c9b4e9a281" parameter_set:<name:"--t" parameter_type:DOUBLE value:"0.6268" > 
2019/06/25 17:44:15 Study: b3e937c9b4e9a281 Suggestions trials:<trial_id:"d9e01b606dce0b7a" study_id:"b3e937c9b4e9a281" parameter_set:<name:"--t" parameter_type:DOUBLE value:"0.4861" > > trials:<trial_id:"q5c9d35a8f988475" study_id:"b3e937c9b4e9a281" parameter_set:<name:"--t" parameter_type:DOUBLE value:"0.6268" > > 
2019/06/25 17:44:16 Fail to Update StudyJob b3e937c9b4e9a281 : Operation cannot be fulfilled on studyjobs.kubeflow.org "mnist": the object has been modified; please apply your changes to the latest version and try again
ERROR: 2019/06/25 17:45:46 pickfirstBalancer: failed to NewSubConn: rpc error: code = Canceled desc = grpc: the client connection is closing
2019/06/25 17:46:16 Study: b3e937c9b4e9a281 CreatedTrials :
2019/06/25 17:46:16 	trial_id:"gc8bdd2735afac95" study_id:"b3e937c9b4e9a281" parameter_set:<name:"--t" parameter_type:DOUBLE value:"0.2261" > 
2019/06/25 17:46:16 	trial_id:"yc959d66eccf96e7" study_id:"b3e937c9b4e9a281" parameter_set:<name:"--t" parameter_type:DOUBLE value:"0.8201" > 
2019/06/25 17:46:16 Study: b3e937c9b4e9a281 Suggestions trials:<trial_id:"gc8bdd2735afac95" study_id:"b3e937c9b4e9a281" parameter_set:<name:"--t" parameter_type:DOUBLE value:"0.2261" > > trials:<trial_id:"yc959d66eccf96e7" study_id:"b3e937c9b4e9a281" parameter_set:<name:"--t" parameter_type:DOUBLE value:"0.8201" > > 
2019/06/25 17:46:17 Fail to Update StudyJob b3e937c9b4e9a281 : Operation cannot be fulfilled on studyjobs.kubeflow.org "mnist": the object has been modified; please apply your changes to the latest version and try again
2019/06/25 17:47:16 Fail to Update StudyJob b3e937c9b4e9a281 : Operation cannot be fulfilled on studyjobs.kubeflow.org "mnist": the object has been modified; please apply your changes to the latest version and try again
2019/06/25 17:47:17 Study b3e937c9b4e9a281 reached the request count. It is completed
```

**Comment by johnugeorge** [comment]
are all katib services running?

**Comment by andreyvelich** [comment]
@warf34 Why did you name  `objectivevaluename: "A"`. 
I believe, it should be `objectivevaluename: "Accuracy"` and you should print the results like this:
```
Accuracy=0.99
```
If you want to collect that metric.

**Comment by warf34** [comment]
> are all katib services running?

Yes, all katib services are running.
But suddenly it works correctly.
After redeploying Minikube, I get the result.
Do you know why?

**Comment by warf34** [comment]
> @warf34 Why did you name `objectivevaluename: "A"` ?
> I believe, it should be `objectivevaluename: "Accuracy"` and you should print the results from your mnist.py like this:
> 
> ```
> Accuracy=0.99
> ```
> 
> If you want to collect that metric.

You are right.
I just decided "A" because this is a test code.

**Comment by andreyvelich** [comment]
@warf34 Metrics Collector will parse metrics from the Training Container only for `objectiveValueName`. In that case, from your first yaml file, it collects nothing from the Training Container. 
You can see, how Metrics Collector works [here](https://www.kubeflow.org/docs/components/hyperparameter/), in Metrics Collector definition.

**Comment by warf34** [comment]
Thanks all!
I have solved the problem.
So, i will close this issue.
